crates/core/src/delta_datafusion/table_provider.rs

```diff
@@ -528,9 +553,17 @@ impl TableProvider for DeltaTable {
         limit: Option<usize>,
     ) -> Result<Arc<dyn ExecutionPlan>> {
         register_store(self.log_store(), session.runtime_env().clone());
+        if let Some(format_options) = &self.file_format_options {
+            format_options.update_session(session)?;
```

Comment:
It might be nice to consolidate this session update with the `register_store` method above.